src/PlaylistController.js

```diff
@@ -0,0 +1,66 @@
+// import assert from 'assert';
+import GenericController from './GenericController';
+import Playlist from './Structs/Playlist';
+
+export default class PlaylistController extends GenericController {
+  constructor(...args) {
+    super(...args);
+    if (window.APPCONTEXT && window.APPCONTEXT.Go && window.APPCONTEXT.Go.h && window.APPCONTEXT.Go.h.length) {
+      this._playlists = Object.assign({}, window.APPCONTEXT.Go.h[0].Hi);
+      this._watchPlaylistObject();
+    } else {
+      this._playlists = {};
+      console.error('Could not listen to playlist changes in the expected location'); // eslint-disable-line
+    }
+
+    this.addMethod('getAll', this.getAll.bind(this));
+    this.addMethod('play', this.play.bind(this));
+    this.addMethod('playWithTrack', this.playWithTrack.bind(this));
+  }
+
+  _navigate(playlist) {
+    return new Promise((resolve) => {
+      window.location.hash = `/pl/${escape(playlist.id)}`;
+      const waitForPage = setInterval(() => {
+        const info = document.querySelector('.material-container-details');
+        if (info && info.querySelector('.title').innerText === playlist.name) {
+          clearInterval(waitForPage);
+          resolve();
+        }
+      }, 10);
+    });
+  }
+
+  _watchPlaylistObject() {
+    const that = this;
+
+    window.APPCONTEXT.Go.h[0].addEventListener('E', () => {
+      this._playlists = Object.assign({}, this._playlists, window.APPCONTEXT.Go.h[0].Hi);
+      that.emitter.emit('change:playlists', this.getAll());
+    });
```

Comment:
> Is there an actual use case for playlists changing?

Yes, check out https://github.com/GPMDP/google-play-music-desktop-remote  Basically when the playlists update we want to update the playlist UI on the remote app.

> Also, the this._playlists logic could easily be relocated into getAll. After all, that.emitter is using getAll itself =/

Yeah I think that was a remnant from before when I was cloning the list to prevent mutation, I then found out GPM does this for us in their code so I stopped :+1: